helix-term/src/ui/popup.rs

```diff
@@ -207,6 +243,67 @@ impl<T: Component> Component for Popup<T> {
                 self.scroll(self.size.1 as usize / 2, false);
                 EventResult::Consumed(None)
             }
+            ctrl!('o') => {
+                let (stringified_body, stringified_buffername, stringified_language) =
+                    if let Some(contents_as_strings) = &self.contents_as_strings {
+                        let stringified_body = contents_as_strings.body.clone();
+                        let stringified_buffername = contents_as_strings.title.clone();
+                        let stringified_language = contents_as_strings.language.clone();
+                        (
+                            stringified_body,
+                            stringified_buffername,
+                            stringified_language,
+                        )
+                    } else {
+                        ("dummy".into(), self.id.into(), "none".into())
+                    };
+
+                // open a new buffer to show the popup content
+                cx.editor.new_file(Action::Replace);
+
+                // TODO: how can we meaningfully stringify the content of a Popup? for now dummy
+                // TODO: is this an un-necessary copy? ...
+                // put the content in the new buffer
+                let (view, doc) = current!(cx.editor);
+                paste_impl(
+                    &[stringified_body],
+                    doc,
+                    view,
+                    Paste::Before,
+                    1,
+                    cx.editor.mode,
+                );
+
+                doc.set_language_by_language_id(
+                    &stringified_language,
+                    cx.editor.syn_loader.clone(),
+                )
+                .unwrap();
+
+                let text = doc.text().slice(..);
+                let selection = doc
+                    .selection(view.id)
+                    .clone()
+                    .transform(|range| range.put_cursor(text, 0, cx.editor.mode == Mode::Select));
+                push_jump(view, doc);
+                doc.set_selection(view.id, selection);
+
+                // put a meaningful title
+                // TODO: get a meaningful buffer title
+                let stringified_buffername = Some(Path::new(&stringified_buffername));
+                doc.set_path(stringified_buffername);
```

Comment:
The path of a buffer isn't meant to be a name - this should be left as a scratch buffer instead

Reply:
No, these docs shouldn't be opened for very long if they're just being used for reference anyways so I think scratch is fine